lib/Parse/ParseType.cpp

```diff
@@ -1376,10 +1375,8 @@ bool Parser::isOptionalToken(const Token &T) const {
   // A postfix or bound infix operator token that begins with '?' can be
   // optional too. We'll munch off the '?', so long as it is left-bound with
   // the type (i.e., parsed as a postfix or unspaced binary operator).
-  if ((T.is(tok::oper_postfix) || T.is(tok::oper_binary_unspaced)) &&
-      T.getText().startswith("?"))
-    return true;
-  return false;
+  return ((T.is(tok::oper_postfix) || T.is(tok::oper_binary_unspaced)) &&
```

Comment:
These long return conditions started off as long conditions. I don't think we gain much by just inlining them straight into return position. Can we split this and its comment text into the components across the `&&` here?

```
// A postfix or bound infix operator token that begins with '?' can be
// optional too.
if (!T.is(tok::oper_postfix) && !T.is(tok::oper_binary_unspaced)) {
  return false;
}
// We'll munch off the '?', so long as it is left-bound with
// the type (i.e., parsed as a postfix or unspaced binary operator).
return T.getText().startswith("?"));
````